pennylane/devices/null_qubit.py

```diff
@@ -509,3 +510,17 @@ def zeros_like(var, shots):
             return math.zeros(var.aval.shape, dtype=var.aval.dtype, like="jax")
 
         return [zeros_like(var, Shots(shots).total_shots) for var in jaxpr.outvars]
+
+
+def _op_has_decomp(op):
+    """Check if an operator has a decomposition, taking into account the graph-based decomposition system.
+
+    Args:
+        op (Operator): The operator to check.
+
+    Returns:
+        bool: True if the operator has a decomposition, False otherwise.
+    """
+    if enabled_graph():
+        return has_decomp(type(op))
```

Comment:
You should be able to use `has_decomp(op)` after https://github.com/PennyLaneAI/pennylane/pull/8286